lsp/jsonnet_ls.lua

```diff
@@ -0,0 +1,37 @@
+--- @brief
+---
+--- https://github.com/grafana/jsonnet-language-server
+---
+--- A Language Server Protocol (LSP) server for Jsonnet.
+---
+--- The language server can be installed with `go`:
+--- ```sh
+--- go install github.com/grafana/jsonnet-language-server@latest
+--- ```
+local util = require 'lspconfig.util'
+
+-- common jsonnet library paths
+local function jsonnet_path(root_dir)
+  local paths = {
+    root_dir .. '/lib',
+    root_dir .. '/vendor',
+  }
+  return table.concat(paths, ':')
+end
+
+local root_dir = function(fname)
+  return util.root_pattern 'jsonnetfile.json'(fname)
+    or vim.fs.dirname(vim.fs.find('.git', { path = fname, upward = true })[1])
+end
+
+return {
+  cmd = { 'jsonnet-language-server' },
+  cmd_env = {
+    JSONNET_PATH = vim.env.JSONNET_PATH or jsonnet_path(root_dir()),
```

Comment:
I'm not sure the fallback here is a good idea. It will be relative to cwd rather than the buffer. It's also calculated once in the beginning. So if you `:cd` around it will be incorrect as well. It will also be wrong when doing something like `nvim ../foo/bar.jsonnet` or just `nvim ~/some/other/dir/file.jsonnet`.